File: Assets/ToluaFramework/Lua/manager/textureManager.lua
--region *.lua
--Date
--此文件由[BabeLua]插件自动生成

local textureManager = {}
local assetType = "texture"

-------------------------------------------------------------------
--
-------------------------------------------------------------------
function textureManager.setup()
    AssetPoolManager.instance:AddPool(assetType, "texture", true)
end

-------------------------------------------------------------------
--
-------------------------------------------------------------------
function textureManager.load(assetPath, assetName)  
    return AssetPoolManager.instance:Alloc(assetType, string.lower(assetPath), string.lower(assetName))
end

-------------------------------------------------------------------
--
-------------------------------------------------------------------
function textureManager.unload(tex)
    local suc = AssetPoolManager.instance:Dealloc(assetType, tex)
end

return textureManager

--endregion

File: Assets/ToluaFramework/Scripts/Utility/TextureManager/TextureManager.cs
using UnityEngine;

public class TextureManager
{
    #region Instance

    /// <summary>
    /// 
    /// </summary>
    private static TextureManager mInstance = new TextureManager();

    /// <summary>
    /// 
    /// </summary>
    public static TextureManager instance
    {
        get { return mInstance; }
    }

    #endregion

    #region Public 

    /// <summary>
    /// 
    /// </summary>
    public void Setup()
    {

    }

    /// <summary>
    /// 
    /// </summary>
    /// <param name="assetPath"></param>
    /// <param name="assetName"></param>
    /// <returns></returns>
    public Texture Load(string assetPath, string assetName)
    {
        return AssetPoolManager.instance.Alloc(AssetPoolManager.Type.Texture, assetPath, assetName) as Texture;
    }

    /// <summary>
    /// 
    /// </summary>
    /// <param name="tex"></param>
    public void Unload(Texture tex)
    {
#if UNITY_EDITOR
        Debug.Assert(tex != null);
#endif

        AssetPoolManager.instance.Dealloc(AssetPoolManager.Type.Texture, tex);
    }

    #endregion
}
